--- /opt/local/php5/lib/php/Auth.php	2009-07-10 14:22:53.000000000 +0200
+++ Auth.php	2009-07-10 16:10:32.000000000 +0200
@@ -223,7 +223,7 @@
      *
      * @var string
      */
-    var $version = "@version@";
+    var $version = "1.6.1";
 
     /**
      * Flag to use advanced security
@@ -313,6 +313,11 @@
      */
     var $regenerateSessionId = false;
 
+    /**
+     * Added by Flux CMS
+     */
+    var $idcol = 'id';
+
     // }}}
     // {{{ Auth() [constructor]
 
@@ -410,6 +415,14 @@
                 $this->_postPassword = $options['postPassword'];
                 unset($options['postPassword']);
             }
+
+            /**
+             * Added by Flux CMS
+             */
+            if (!empty($options['idcol'])) {
+                $this->idcol = $options['idcol'];
+            }
+
             if (isset($options['advancedsecurity'])) {
                 $this->advancedsecurity = $options['advancedsecurity'];
                 unset($options['advancedsecurity']);
@@ -1114,6 +1127,17 @@
         return('');
     }
 
+    /*
+     * Added by Flux CMS
+     */
+    function getUserId()
+    {
+        if (!isset($this->session['username'])) {
+            return '';
+        }
+        return $this->session['data'][$this->idcol];
+    }
+
     // }}}
     // {{{ getStatus()
 
--- /opt/local/php5/lib/php/Auth/Container/MDB2.php	2009-07-10 14:22:53.000000000 +0200
+++ Auth/Container/MDB2.php	2009-07-10 16:11:51.000000000 +0200
@@ -141,10 +141,19 @@
             }
             $this->options['final_usernamecol'] = $this->db->quoteIdentifier($this->options['usernamecol'], true);
             $this->options['final_passwordcol'] = $this->db->quoteIdentifier($this->options['passwordcol'], true);
+            /**
+             * Added by Flux CMS
+             */
+            $this->options['final_idcol'] = $this->db->quoteIdentifier($this->options['idcol'], true);
         } else {
             $this->options['final_table'] = $this->options['table'];
             $this->options['final_usernamecol'] = $this->options['usernamecol'];
             $this->options['final_passwordcol'] = $this->options['passwordcol'];
+            /**
+             * Added by Flux CMS
+             */
+            $this->options['final_idcol'] = $this->options['idcol'];
+
         }
 
         return true;
@@ -209,6 +218,10 @@
         $this->options['table']       = 'auth';
         $this->options['usernamecol'] = 'username';
         $this->options['passwordcol'] = 'password';
+        /**
+         * Added by Flux CMS
+         */
+        $this->options['idcol'] = 'id';
         $this->options['dsn']         = '';
         $this->options['db_fields']   = '';
         $this->options['cryptType']   = 'md5';
@@ -306,7 +319,11 @@
             $sql_from = '*';
         } else {
             $sql_from = $this->options['final_usernamecol'].
-                ", ".$this->options['final_passwordcol'];
+                ", ".$this->options['final_passwordcol'].
+            /**
+             * Added by Flux CMS
+             */
+                ", ".$this->options['final_idcol'];
 
             if (strlen($fields = $this->_quoteDBFields()) > 0) {
                 $sql_from .= ', '.$fields;
